### PR TITLE
* FIX [parquet] When initializing the parquet file queue, remove any …

### PR DESCRIPTION
**Pull request by RanMaoyi**
…parquet files that have not yet been written

### CODE REVIEW

File: src/supplemental/nanolib/parquet/parquet.cc [hunk 1]
```diff
@@ -688,11 +702,10 @@ parquet_file_queue_init(conf_parquet *conf)
 						         "file count: %d.",
 						    conf->file_count);
 					} else {
+						free(file_path);
```
Comment (xinyi-xs) [inline src/supplemental/nanolib/parquet/parquet.cc:705]: This could result in a double free situation.

**Review by xinyi-xs — CHANGES_REQUESTED** [review]

File: src/supplemental/nanolib/parquet/parquet.cc [hunk 2]
```diff
@@ -680,6 +680,20 @@ parquet_file_queue_init(conf_parquet *conf)
...
 				}
+				sprintf(file_path, "%s/%s", conf->dir,
+				    ent->d_name);
+
+				if (strstr(ent->d_name, "_") == NULL) {
+					if (unlink(file_path) != 0) {
+						log_error("Failed to remove file %s errno: %d",
+						    file_path, errno);
+					} else {
+						log_warn("Found a file without md5sum, "
```
Comment (wanghaEMQ) [inline src/supplemental/nanolib/parquet/parquet.cc:691]: Why there will be a file without md5sum?

File: src/supplemental/nanolib/parquet/parquet.cc (hunk 1 above)
Comment (RanMaoyi) [reply src/supplemental/nanolib/parquet/parquet.cc:705]: don't understand, file_path not enqueue parquet_file_queue, where will cause double free?

File: src/supplemental/nanolib/parquet/parquet.cc (hunk 2 above)
Comment (RanMaoyi) [reply src/supplemental/nanolib/parquet/parquet.cc:691]: parquet writes to a file by first creating a file without md5 and then renaming it to a file with md5 after the write is complete, so if nanomq quits while the file is being written, the file without md5 will be produced.

File: src/supplemental/nanolib/parquet/parquet.cc (hunk 1 above)
Comment (xinyi-xs) [reply src/supplemental/nanolib/parquet/parquet.cc:705]: See function remove_old_file(void).

Comment (RanMaoyi) [reply src/supplemental/nanolib/parquet/parquet.cc:705]: But this file_path was created by alloc and is not queued yet. The file_path in remove_old_file() is taken from the queue.

**Pull request merged**